src/Cli/Microsoft.TemplateEngine.Cli/TemplatePackageCoordinator.cs

```diff
@@ -207,9 +207,21 @@ internal async Task<NewCommandStatus> EnterInstallFlowAsync(InstallCommandArgs a
 
             foreach (string installArg in args.TemplatePackages)
             {
-                string[] splitByColons = installArg.Split(new[] { "::" }, StringSplitOptions.RemoveEmptyEntries);
-                string identifier = splitByColons[0];
-                string? version = splitByColons.Length > 1 ? splitByColons[1] : null;
+                string[] splitByColons = installArg.Split(["::"], StringSplitOptions.RemoveEmptyEntries);
+                string[] splitByAt = installArg.Split('@', StringSplitOptions.RemoveEmptyEntries);
+                string[] split = splitByColons.Length > splitByAt.Length ? splitByColons :
+                                 splitByAt.Length > splitByColons.Length ? splitByAt :
+                                 splitByColons[0].Length < splitByAt[0].Length ? splitByColons :
```

Comment:
What scenario is this condition for?

Reply:
This was supposed to be to handle cases in which the separator is ::, the separator is @, and there isn't a real separator; it just ends in one of those. I don't think that last case is common, but it appeared in a test, so I made it complicated for that reason.

I think I should change it to this, though:
string[] split = installArg.Split("::", StringSplitOptions.RemoveEmptyEntries).SelectMany(arg => arg.Split('@', StringSplitOptions.RemoveEmptyEntries)).ToArray();

Seems simpler, and it should be robust to another edge case that I don't think is super important:
dotnet new install package1::version1 package2@version2